src/rust/engine/options/src/config.rs

```diff
@@ -216,133 +215,16 @@ fn toml_table_to_dict(table: &Value) -> HashMap<String, Val> {
     }
 }
 
-#[derive(Clone)]
-struct ConfigSource {
```

Comment:
We get rid of ConfigSource entirely, and fold its functionality directly into `Config` below, since they are now one-to-one.